src/main/kotlin/com/nexters/gamss/notification/service/CardCreatedNotifier.kt

```
@@ -0,0 +1,67 @@
+     *
+     * **대가를 알고 쓴다: 이 예외는 배치 루프를 중단시킨다.** 부르는 쪽에 catch 가 없어서 남은 방들은
+     * 그날 종료도 카드 생성도 되지 않는다(3개 중 1개만 처리되는 것을 실제로 확인했다). 그래도 계속
+     * 도는 것보다 낫다고 봤다 — 가드가 발동했다는 것은 배치가 트랜잭션 안에 있다는 뜻이고, 그 상태로
+     * 끝까지 돌면 DB 커넥션을 붙잡은 채 LLM 을 수백 번 부른다.
+     */
+    fun notifyCardCreated(memberId: Long): PushSendResult =
+        try {
+            notifier.send(listOf(memberId), MESSAGE)
+        } catch (e: PushInTransactionException) {
+            throw e
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)